pkg/ccn-lite/Makefile

```diff
@@ -7,8 +7,18 @@ PKG_LICENSE=ISC
 
 export RIOT_CFLAGS = $(CFLAGS) $(INCLUDES)
 
-all: git-download
-	cd $(PKG_BUILDDIR)/src && cmake -DCCNL_RIOT=1 -DRIOT_CFLAGS="$(RIOT_CFLAGS)" -DBUILD_TESTING=OFF . && make
-	cp $(PKG_BUILDDIR)/src/lib/libccnl-riot.a $(BINDIR)/ccn-lite.a
+TOOLCHAIN_FILE=$(PKG_BUILDDIR)/xcompile-toolchain.cmake
+
+all: $(PKG_BUILDDIR)/src/Makefile
+	$(MAKE) -C $(PKG_BUILDDIR)/src && \
+	cp $(PKG_BUILDDIR)/src/lib/libccnl-riot.a ${BINDIR}/ccn-lite.a
+
+$(PKG_BUILDDIR)/src/Makefile: $(TOOLCHAIN_FILE)
+	cd $(PKG_BUILDDIR)/src && \
+	cmake -DCMAKE_TOOLCHAIN_FILE=$(TOOLCHAIN_FILE) \
+		  -DCCNL_RIOT=1 -DRIOT_CFLAGS="${RIOT_CFLAGS}" -DBUILD_TESTING=OFF .
+
+$(TOOLCHAIN_FILE): git-download
+	$(RIOTBASE)/dist/tools/cmake/generate-xcompile-toolchain.sh > $(TOOLCHAIN_FILE)
```

Comment:
```$(RIOTBASE)/dist/tools/cmake/generate-xcompile-toolchain.sh > $@```

Reply:
ah, Makefile magic - kind of :) will add

Reply:
mhm, doesn't work though - `$@` is expanded to `/xcompile-toolchain.cmake` only, without pkg build dir set.

Reply:
so is ```$(TOOLCHAIN_FILE)```? ```$@``` should evaluate to the target name.

Reply:
I'm not sure whats going wrong fact is:

```
$(RIOTBASE)/dist/tools/cmake/generate-xcompile-toolchain.sh > $@
```

is expanded to 
```
/path/to/RIOT/dist/tools/cmake/generate-xcompile-toolchain.sh > /xcompile-toolchain.cmake
```

which obviously gives an error, because it is not allow to write to my root dir.

Reply:
so the pkgbuilddir is empty at this point

Reply:
leaving it as is, works

Reply:
ok, thanks for trying. Weird that it can match the target but doesn't evaluate $@ afterwards?

Reply:
At that time `PKG_BUILDDIR` is not defined because `pkg.mk` has not been included.
Goals are evaluated immediately not deferred. So the rule is really `/xcompile-toolchain.cmake`.

But the problem was also in the `pkg/relic/Makefile` before, so would require a real fix that could come later.